app/backend/src/ui/methods/task/update-task.ts

```diff
@@ -10,5 +11,20 @@ import TaskModel from '../../../database/models/task';
 export async function updateTask(data: UpdateTaskPayload): Promise<Task | null> {
   const query = { _id: data._id };
 
-  return TaskModel.findOneAndUpdate(query, data).exec();
+  const task = await TaskModel.findOneAndUpdate(query, data, { new: true }).exec();
+
+  if (!task || !task.statusId) {
+    return null;
+  }
+  // Add expanded status data (if any) to result object
+  const status = await getStatusById(task.statusId);
+
+  if (!status) {
+    return task.toObject();
+  }
+
+  return {
```

Comment:
why there is a difference? I think, the response format should not depend on whether the `status` is defined or not

Reply:
fixed